src/main/kotlin/com/zepben/auth/client/AuthProviderConfig.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2024 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.auth.client
+
+import com.zepben.auth.common.AuthException
+import com.zepben.auth.common.AuthMethod
+import com.zepben.auth.common.StatusCode
+import io.vertx.core.json.DecodeException
+import io.vertx.core.json.Json
+import io.vertx.core.json.JsonObject
+import java.io.IOException
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+
+data class ProviderDetails(
+    val issClaim: String,
+    val tokenEndpoint: String,
+    val jwkUrl: String
+)
+
+data class AuthProviderConfig(
+    val issuer: String,
+    val audience: String,
+    val authType: AuthMethod = AuthMethod.OAUTH,
```

Comment:
Are we removing EntraID and Auth0 from the AuthMethod enum?

Reply:
as per our conversation, yes, we will but as a separate step.